File: sdks/javascript/cerberus-ai.ts
/**
 * Cerberus AI JavaScript SDK
 * 
 * Official JavaScript/TypeScript client for Cerberus AI API
 */

interface Message {
  role: string;
  content: string;
}

interface ChatCompletionOptions {
  model?: string;
  temperature?: number;
  maxTokens?: number;
  debugMode?: boolean;
}

interface CodeAnalyzeOptions {
  checks?: string[];
}

interface CodeRefactorOptions {
  goals?: string[];
}

export class CerberusAI {
  private apiKey: string;
  private baseUrl: string;

  constructor(apiKey: string, baseUrl: string = 'http://localhost:8000') {
    this.apiKey = apiKey;
    this.baseUrl = baseUrl.replace(/\/$/, '');
  }

  private async request(endpoint: string, options: RequestInit = {}): Promise<any> {
    const response = await fetch(`${this.baseUrl}${endpoint}`, {
      ...options,
      headers: {
        'Authorization': `Bearer ${this.apiKey}`,
        'Content-Type': 'application/json',
        ...options.headers,
      },
    });

    if (!response.ok) {
      throw new Error(`API Error: ${response.status} ${response.statusText}`);
    }

    return response.json();
  }

  /**
   * Create a chat completion
   */
  async chatCompletion(
    messages: Message[],
    options: ChatCompletionOptions = {}
  ): Promise<any> {
    return this.request('/v1/chat/completions', {
      method: 'POST',
      body: JSON.stringify({
        messages,
        model: options.model || 'cerberus-pro',
        temperature: options.temperature || 0.7,
        max_tokens: options.maxTokens || 2048,
        debug_mode: options.debugMode || false,
      }),
    });
  }

  /**
   * Analyze code for issues
   */
  async analyzeCode(
    code: string,
    language: string,
    options: CodeAnalyzeOptions = {}
  ): Promise<any> {
    return this.request('/v1/code/analyze', {
      method: 'POST',
      body: JSON.stringify({
        code,
        language,
        checks: options.checks || ['security', 'performance', 'style'],
      }),
    });
  }

  /**
   * Debug code with error
   */
  async debugCode(
    error: string,
    code: string,
    language: string,
    context?: string
  ): Promise<any> {
    return this.request('/v1/code/debug', {
      method: 'POST',
      body: JSON.stringify({
        error,
        code,
        language,
        context,
      }),
    });
  }

  /**
   * Refactor code
   */
  async refactorCode(
    code: string,
    language: string,
    options: CodeRefactorOptions = {}
  ): Promise<any> {
    return this.request('/v1/code/refactor', {
      method: 'POST',
      body: JSON.stringify({
        code,
        language,
        goals: options.goals || ['readability', 'maintainability'],
      }),
    });
  }

  /**
   * List available models
   */
  async listModels(): Promise<any> {
    return this.request('/v1/models', { method: 'GET' });
  }

  /**
   * Get usage statistics
   */
  async getUsage(): Promise<any> {
    return this.request('/v1/usage', { method: 'GET' });
  }
}

export default CerberusAI;

File: sdks/python/cerberus_ai.py
"""
Cerberus AI Python SDK

Official Python client for Cerberus AI API
"""

import requests
from typing import List, Dict, Optional, Any


class CerberusAI:
    """Cerberus AI API Client"""
    
    def __init__(self, api_key: str, base_url: str = "http://localhost:8000"):
        self.api_key = api_key
        self.base_url = base_url.rstrip("/")
        self.session = requests.Session()
        self.session.headers.update({
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json"
        })
    
    def chat_completion(
        self,
        messages: List[Dict[str, str]],
        model: str = "cerberus-pro",
        temperature: float = 0.7,
        max_tokens: int = 2048,
        debug_mode: bool = False
    ) -> Dict[str, Any]:
        """
        Create a chat completion
        
        Args:
            messages: List of message dicts with 'role' and 'content'
            model: Model to use (cerberus-lite, cerberus-pro, cerberus-debug)
            temperature: Sampling temperature (0-1)
            max_tokens: Maximum tokens to generate
            debug_mode: Enable debug mode for deep analysis
        
        Returns:
            Response dict with completion
        """
        response = self.session.post(
            f"{self.base_url}/v1/chat/completions",
            json={
                "model": model,
                "messages": messages,
                "temperature": temperature,
                "max_tokens": max_tokens,
                "debug_mode": debug_mode
            }
        )
        response.raise_for_status()
        return response.json()
    
    def analyze_code(
        self,
        code: str,
        language: str,
        checks: Optional[List[str]] = None
    ) -> Dict[str, Any]:
        """
        Analyze code for issues
        
        Args:
            code: Code to analyze
            language: Programming language
            checks: List of checks (security, performance, style)
        
        Returns:
            Analysis results
        """
        response = self.session.post(
            f"{self.base_url}/v1/code/analyze",
            json={
                "code": code,
                "language": language,
                "checks": checks or ["security", "performance", "style"]
            }
        )
        response.raise_for_status()
        return response.json()
    
    def debug_code(
        self,
        error: str,
        code: str,
        language: str,
        context: Optional[str] = None
    ) -> Dict[str, Any]:
        """
        Debug code with error
        
        Args:
            error: Error message
            code: Code that caused error
            language: Programming language
            context: Additional context
        
        Returns:
            Debug information with solutions
        """
        response = self.session.post(
            f"{self.base_url}/v1/code/debug",
            json={
                "error": error,
                "code": code,
                "language": language,
                "context": context
            }
        )
        response.raise_for_status()
        return response.json()
    
    def refactor_code(
        self,
        code: str,
        language: str,
        goals: Optional[List[str]] = None
    ) -> Dict[str, Any]:
        """
        Refactor code
        
        Args:
            code: Code to refactor
            language: Programming language
            goals: Refactoring goals (readability, maintainability)
        
        Returns:
            Refactored code and explanation
        """
        response = self.session.post(
            f"{self.base_url}/v1/code/refactor",
            json={
                "code": code,
                "language": language,
                "goals": goals or ["readability", "maintainability"]
            }
        )
        response.raise_for_status()
        return response.json()
    
    def list_models(self) -> Dict[str, Any]:
        """List available models"""
        response = self.session.get(f"{self.base_url}/v1/models")
        response.raise_for_status()
        return response.json()
    
    def get_usage(self) -> Dict[str, Any]:
        """Get usage statistics"""
        response = self.session.get(f"{self.base_url}/v1/usage")
        response.raise_for_status()
        return response.json()
